src/sql/magic_cmd.py

```diff
@@ -35,15 +35,39 @@ def error(self, message):
 class SqlCmdMagic(Magics, Configurable):
     """%sqlcmd magic"""
 
+
     @line_magic("sqlcmd")
     @magic_arguments()
     @argument("line", default="", type=str, help="Command name")
-    def execute(self, line="", cell=""):
+    def _validate_inputs(self, line=""):
+        """
+        Command
+        """
+
+        if line == "":
+            raise UsageError(
+                "Missing argument for %sqlcmd. "
+                "Valid commands are: 'tables', 'columns', 'profile'"
+            )
+        else:
+            split = arg_split(line)
+            command, others = split[0].strip(), split[1:]
+
+            if command == "tables" or command == "columns" or command == "test" or command == "profile":
+                return self.execute(command, others)
+            else:
```

Comment:
You can maintain a available command list and check if the input command in the list
```suggestion
    AVAILABLE_SQLCMD_COMMANDS = ["tables", "columns", "test", "profile"]
    if command in AVAILABLE_SQLCMD_COMMANDS:
        return self.execute(command, others)
```